homeassistant/components/husqvarna_automower_ble/coordinator.py

```diff
@@ -0,0 +1,127 @@
+"""Provides the DataUpdateCoordinator."""
+
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from automower_ble.mower import Mower
+from bleak import BleakError
+from bleak_retry_connector import close_stale_connections_by_address
+
+from homeassistant.components import bluetooth
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+    UpdateFailed,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(seconds=60)
+
+
+class HusqvarnaCoordinator(DataUpdateCoordinator[dict[str, bytes]]):
+    """Class to manage fetching data."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        logger: logging.Logger,
+        mower: Mower,
+        device_info: DeviceInfo,
+        address: str,
+        model: str,
+    ) -> None:
+        """Initialize global data updater."""
+        super().__init__(
+            hass=hass,
+            logger=logger,
+            name="Husqvarna Automower BLE Data Update Coordinator",
```

Comment:
Based on other examples, this is usually = DOMAIN (not sure if it really matters)